cmd/zdb/zdb.c

```diff
@@ -6846,6 +6873,16 @@ main(int argc, char **argv)
 		(void) fprintf(stderr, "-p option requires use of -e\n");
 		usage();
 	}
+	if (dump_opt['d']) {
+		/* <poolname>/<decimal> or 0x<hex> is accepted */
+		if (argv[2]) {
+			if ((objset_str = strchr(argv[2], '/')) != NULL &&
+			    objset_str++ != NULL && isdigit(objset_str[0])) {
```

Comment:
Rather then checking for a leading digit can we not rely on `strtoull()` with a zero base to parse the objset_str.  When `endptr` is specified it's possible to determine if either a string or valid number where provided.